ibm/service/uko/data_source_ibm_key_template.go

```diff
@@ -0,0 +1,311 @@
+// Copyright IBM Corp. 2022 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package uko
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/flex"
+	"github.com/IBM/ibm-hpcs-uko-sdk/ukov4"
+)
+
+func DataSourceIbmKeyTemplate() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: DataSourceIbmKeyTemplateRead,
+
+		Schema: map[string]*schema.Schema{
+			"instance_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The ID of the UKO instance this resource exists in.",
+			},
+			"region": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The region of the UKO instance this resource exists in.",
+			},
+			"id": &schema.Schema{
```

Comment:
"id" reserved for Terraform core to identity and resource /datasource in state file management. defien tempalte_id and vault_id as arguments